### ISSUE TITLE
extend setLog to allow disabling removeColors call

### ISSUE THREAD

**Issue by cognisent** [vitaly-t/pg-monitor#24]
We really like the way logs are formatted by **pg-monitor**, but we also need to customize the logger. Currently, if you provide a custom log function, it will always call `removeColors` before it passes the message in. We want the colors though! We just also want to customize logging.

I played around with some options, and I wanted to get your thoughts to see if I should submit a PR.

I could change `setLog` to allow it to take either of:
* `log: function(msg, info)` _Current_
* `options: { log: function(msg, info), removeColors: Boolean }` _New!_

Setting `options.removeColors` to `false` could result in either:
* `msg` is colored **-but-** `info.text` is plain
* `msg` is colored **-and-** `info.text` is colored

Already, the two are _slightly_ different in that `info.text` has `.trim()` called on the string, whereas `msg` does not; so, I'm not sure if you think they should also be differentiated in this way as well.

Usage would change to be able to look like this:
```js
monitor.setLog({
    removeColors: false,
    log(msg, info) { /* ... */ }
});
```

This allows the best of both worlds, with output like this:   
![image](https://user-images.githubusercontent.com/1672405/66609502-3b543f00-eb87-11e9-95b7-c0e7904e39d9.png)

What do you think?

**Comment by vitaly-t** [comment]
What is it that you are trying to change in the output? - replace timestamp with `[PG]`?

But why? What is it that the database log is lacking? I'd like to see real a practical example to understand the benefit of the change.

**Comment by vitaly-t** [comment]
B.T.W. [This](https://github.com/vitaly-t/pg-monitor/issues/25) will present an opportunity for all kinds of changes.

**Comment by cognisent** [comment]
Hey, thanks for the reply. The screenshot was just an example. We don't need to change the output, we just want to customize the logging (so that we can use something other than `console.log`) but not lose the colors that you set up, like I said, here:

>Currently, if you provide a custom log function, it will always call `removeColors` before it passes the message in. We want the colors though! We just also want to customize logging.

**Comment by vitaly-t** [comment]
@nazrhyn At this point I think I will back-log this to #25

**Comment by cognisent** [comment]
@vitaly-t That's fine. We can maintain our local fork until then. When you rewrite it though, whatever your replacement to this section is, please make it so that the call to `removeColors` can be turned off!

https://github.com/vitaly-t/pg-monitor/blob/94fc32690a782c5d1711ed7124d90e5109e9e8f1/lib/index.js#L433-L445

Thanks for your time and for your work. You have some of the best-documented, most-sensibly-organized libraries in the Javascript world. We appreciate it. ❤️

**Comment by vitaly-t** [comment]
>  please make it so that the call to `removeColors`

Early to say, but most likely I will include both the original, and without colors at the same time.

The whole thing will likely be rewritten, given the opportunity of making all the breaking changes.

**Comment by vitaly-t** [comment]
I couldn't find enough time to do the conversion, and so instead, I added `colorText` property into the `info` structure, and released it in [v1.4.0](https://github.com/vitaly-t/pg-monitor/releases/tag/1.4.0).

@nazrhyn  That should be enough, I think ;)

**Comment by cognisent** [comment]
Awesome! Thanks for the update and for tracking this. 🎉
